### ISSUE TITLE
Feature Request: add support for error bars in graphs

### ISSUE THREAD

**Issue by daverodgman** [grafana/grafana#6106]
- **I'm submitting a ...**
- [*] Feature request

Please include this information:
- What Grafana version are you using? 3.1.0
- What datasource are you using? InfluxDB
- What OS are you running grafana on? Ubuntu 14.04

Please add support for error bars in graphs. The ability to show configurable size intervals from standard deviation and number of samples (i.e., get these two numbers from additional fields in the point)  would be great.

**Comment by harshitha-m** [comment]
It would be great if this feature is added

**Comment by appmagnetics** [comment]
I'm also voting for this feature. This is widely used in statistics / machine learning.

**Comment by huy** [comment]
👍

**Comment by ColinIanKing** [comment]
This would make the graphing actually useful. At the moment it's a feature that is really lacking

**Comment by hildogjr** [comment]
I had just posted in the community and no answer. Now I realize that still as request.
https://community.grafana.com/t/min-max-and-value-in-line-graph/36285

I would like to specif max and min of an already defined line and get those bellow kind of graphs:
![image](https://user-images.githubusercontent.com/2211474/92641627-68d4cb80-f2b5-11ea-8f07-1988211622c3.png)

![image](https://user-images.githubusercontent.com/2211474/92641641-6e321600-f2b5-11ea-934a-c2036732113b.png)

**Comment by lenaxia** [comment]
+1 on this

**Comment by carlosandre-fernandes** [comment]
+1

**Comment by frank0434** [comment]
+1

**Comment by Dmitry-NC** [comment]
+1 Someone checked how it works https://github.com/Madhu94/grafana-highcharts ?

**Comment by alflexRH** [comment]
+1

**Comment by anilpraneeth** [comment]
+1

**Comment by RAYs3T** [comment]
+1

**Comment by alkanen** [comment]
Yes please!

**Comment by castillo92** [comment]
+1

Something like this

![Chart max min](https://powerbidocs.com/wp-content/uploads/2021/03/Add-max-min-line-into-line-visuals.png)

**Comment by hildogjr** [comment]
This alterar different, more like a "simple" visualization tool. The old comments appear to ask by a graph with double/tripple query: one for the main line, one or two others to the max/min valeu (wrap limits) or deviation.

At least this was my intention.

**Comment by chris-hoe** [comment]
> This alterar different, more like a "simple" visualization tool. The old comments appear to ask by a graph with double/tripple query: one for the main line, one or two others to the max/min valeu (wrap limits) or deviation.
> 
> At least this was my intention.

+1 for this - it's a feature I'm seriously missing since changing from rrd to influxdb/grafana!

**Comment by CDaxi** [comment]
From my point of view, this issue can be closed.

This feature is possible already (at least now) without any special kind of visualization using "timeseries" visualization.
You will need three series:
* Min
* Mean
* Max

In flux (influxdb2) you can use a query like:
```
start = v.timeRangeStart
stop = v.timeRangeStop
every = v.windowPeriod

data = from(bucket: "homeassistant")
  |> range(start: start, stop: stop)
  |> filter(fn: (r) => r._measurement == "sensor.boiler_outside_temperature" and  r._field == "value")

maxValue = data
  |> aggregateWindow(every: every, fn: max, createEmpty: true)
  |> keep(columns: ["_time", "entity_id", "_field", "_value"])
  |> map(fn: (r) => ({r with aggr: "max"}))
minValue = data
  |> aggregateWindow(every: every, fn: min, createEmpty: true)
  |> keep(columns: ["_time", "entity_id", "_field", "_value"])
  |> map(fn: (r) => ({r with aggr: "min"}))
meanValue = data
  |> aggregateWindow(every: every, fn: mean, createEmpty: true)
  |> keep(columns: ["_time", "entity_id", "_field", "_value"])
  |> map(fn: (r) => ({r with aggr: "mean"}))

union(tables: [maxValue, minValue, meanValue])
  |> pivot(rowKey: ["_time", "entity_id", "_field"], columnKey: ["aggr"], valueColumn: "_value")
```

Then you can organize the result series order in the Transformations configuration:
<img width="1293" alt="Bildschirmfoto 2023-06-21 um 16 27 25" src="https://github.com/grafana/grafana/assets/2583954/dafc8420-693d-46a6-b8e6-38b26d53e060">

And add some overrides to show only the mean line as a line and fill the space between max and min with a opacity of 10:
<img width="391" alt="Bildschirmfoto 2023-06-21 um 16 27 35" src="https://github.com/grafana/grafana/assets/2583954/ae24eb21-4082-4541-a048-04229b97403d">
<img width="392" alt="Bildschirmfoto 2023-06-21 um 16 27 42" src="https://github.com/grafana/grafana/assets/2583954/f6f5efd9-9718-4d64-8d22-1caed4dd5b20">
<img width="392" alt="Bildschirmfoto 2023-06-21 um 16 27 51" src="https://github.com/grafana/grafana/assets/2583954/30f2396d-5be4-4834-b2d5-05f7d60dad72">


The result will look like this:
<img width="1315" alt="Bildschirmfoto 2023-06-21 um 16 31 02" src="https://github.com/grafana/grafana/assets/2583954/4f7da3b3-e89e-41f9-96de-ddcac59df5c7">

And the tooltip displays all series in the order you have selected in transformation configuration:
<img width="315" alt="Bildschirmfoto 2023-06-21 um 16 31 08" src="https://github.com/grafana/grafana/assets/2583954/ec19b3cc-af10-4d04-bf8e-aefa474682ed">

To display all series in the tooltip, select the options as below:
<img width="394" alt="Bildschirmfoto 2023-06-21 um 16 35 15" src="https://github.com/grafana/grafana/assets/2583954/699ace61-5a0a-4f1d-8df2-d6ed14d7bb38">

**Comment by BrainStone** [comment]
I disagree. Mainly because this requires that you have a fixed amount of time series and you need to add this for every single one of them.  
And then on top of it you need to know what you're doing to get that result. It's not obvious at all how to do that.

**Comment by leeoniya** [comment]
> I disagree. Mainly because this requires that you have a fixed amount of time series and you need to add this for every single one of them.

@BrainStone can you elaborate on this, please?

would you be able to attach a [panel debug dashboard](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) with your [sanitized] dataset with multiple series that need error bars, and the manually-configured overrides as shown in https://github.com/grafana/grafana/issues/6106#issuecomment-1600960176 that demonstrate the issue?

i suspect part of the solution to automating this is adding support for a proper FieldMatcher for the `fillBelowTo` override, so it can dynamically target not just the upper series, but also the lower series. this gets trickier because you need to be able to use some subset of the upper series name to match the lower series name.

> And then on top of it you need to know what you're doing to get that result. It's not obvious at all how to do that.

the thing is, Grafana provides generic tools for a vast diversity of data formats and requirements. usually when people ask for a "simple" way to do something, they have their very specific/simple use case in mind that may in fact be amenable to a more streamlined/simplified UI. they can completely disregard other people's scenarios and claim that getting things done for their scenario takes 10 clicks instead of one. unfortunately, we don't have the luxury of ignoring every other scenario and then creating a UI that only accommodates one. that's the cost of broad compatibility in some cases, but we're always looking to improve the experience for very common scenarios.

we now have [Window functions](https://github.com/grafana/grafana/pull/77029) in our transformations, which may reduce the need to do additional queries for the upper and lower error bars, e.g. moving avg.
